tensorflow/python/util/nest.py

```diff
@@ -128,12 +137,20 @@ def _sequence_like(instance, args):
   Returns:
     `args` with the type of `instance`.
   """
-  if _is_mapping(instance):
+  if _is_mutable_mapping(instance):
     # Pack dictionaries in a deterministic order by sorting the keys.
     # Notice this means that we ignore the original order of `OrderedDict`
     # instances. This is intentional, to avoid potential bugs caused by mixing
     # ordered and plain dicts (e.g., flattening a dict but using a
     # corresponding `OrderedDict` to pack it back).
+    result = dict(zip(_sorted(instance), args))
+    instance_type = type(instance)
+    if instance_type == _collections.defaultdict:
```

Comment:
You're missing the else branch here (and also for default dicts the constructor needs to be called with the factory argument, as in the old code).

Reply:
You mean to use `_collections.defaultdict(instance.default_factory)` for the `defaultdict` constructor arguement as done in `_is_mapping` branch?

Reply:
Yes.

Reply:
Ok